{
    "fix": "https://github.com/OSGeo/gdal/commit/c368014547ab9f9d7335254c02a87e08a156942a",
    "verify": "0",
    "localId": 8963,
    "project": "gdal",
    "fuzzer": "libfuzzer",
    "sanitizer": "asan",
    "crash_type": "Container-overflow WRITE 8",
    "severity": "High",
    "report": {
        "comments": [
            {
                "projectName": "oss-fuzz",
                "localId": 8963,
                "commenter": {
                    "userId": "382749006",
                    "displayName": "ClusterFuzz-External"
                },
                "timestamp": 1529374720,
                "content": "Detailed report: https://oss-fuzz.com/testcase?key=5198634147643392\n\nProject: gdal\nFuzzer: libFuzzer_gdal_vector_translate_fuzzer\nFuzz target binary: gdal_vector_translate_fuzzer\nJob Type: libfuzzer_asan_gdal\nPlatform Id: linux\n\nCrash Type: Container-overflow WRITE 8\nCrash Address: 0x611000004088\nCrash State:\n  std::__1::vector<double, std::__1::allocator<double> >::__append\n  PCIDSK::ProjParmsFromText\n  PCIDSK::CPCIDSKVectorSegment::GetProjection\n  \nSanitizer: address (ASAN)\n\nRecommended Security Severity: High\n\nRegressed: https://oss-fuzz.com/revisions?job=libfuzzer_asan_gdal&range=201806180619:201806182015\n\nReproducer Testcase: https://oss-fuzz.com/download?testcase_id=5198634147643392\n\nIssue filed automatically.\n\nSee https://github.com/google/oss-fuzz/blob/master/docs/reproducing.md for more information.\n\nThis bug is subject to a 90 day disclosure deadline. If 90 days elapse\nwithout an upstream patch, then the bug report will automatically\nbecome visible to the public.\n\nWhen you fix this bug, please\n  * mention the fix revision(s).\n  * state whether the bug was a short-lived regression or an old bug in any stable releases.\n  * add any other useful information.\nThis information can help downstream consumers.\n\nIf you need to contact the OSS-Fuzz team with a question, concern, or any other feedback, please file an issue at https://github.com/google/oss-fuzz/issues.",
                "descriptionNum": 1
            },
            {
                "projectName": "oss-fuzz",
                "localId": 8963,
                "sequenceNum": 1,
                "commenter": {
                    "userId": "382749006",
                    "displayName": "ClusterFuzz-External"
                },
                "timestamp": 1529375317,
                "amendments": [
                    {
                        "fieldName": "Labels",
                        "newOrDeltaValue": "OS-Linux"
                    }
                ]
            },
            {
                "projectName": "oss-fuzz",
                "localId": 8963,
                "sequenceNum": 2,
                "commenter": {
                    "userId": "3393638884",
                    "displayName": "even....@gmail.com"
                },
                "timestamp": 1529417600,
                "content": "Cannot reproduce. This is apparently an error on a std::vector<> resize() call. That doesn't make sense."
            },
            {
                "projectName": "oss-fuzz",
                "localId": 8963,
                "sequenceNum": 3,
                "commenter": {
                    "userId": "3393638884",
                    "displayName": "even....@gmail.com"
                },
                "timestamp": 1529608842,
                "content": "I'm confident this is a false positive. I coud indeed see the warning in a local Docker, but could make it disappear with this workaround : https://github.com/OSGeo/gdal/commit/c368014547ab9f9d7335254c02a87e08a156942a\r\n\r\nIn this case, the vector capacity is 17 before the resize\r\nTrying a simple reproducer however in the Docker (creating a vector of double with 17 push_back and resize(18) didn't trigger the warning, so this is somewhat subtle"
            },
            {
                "projectName": "oss-fuzz",
                "localId": 8963,
                "sequenceNum": 4,
                "commenter": {
                    "userId": "382749006",
                    "displayName": "ClusterFuzz-External"
                },
                "timestamp": 1529651240,
                "content": "ClusterFuzz has detected this issue as fixed in range 201806210630:201806220629.\n\nDetailed report: https://oss-fuzz.com/testcase?key=5198634147643392\n\nProject: gdal\nFuzzer: libFuzzer_gdal_vector_translate_fuzzer\nFuzz target binary: gdal_vector_translate_fuzzer\nJob Type: libfuzzer_asan_gdal\nPlatform Id: linux\n\nCrash Type: Container-overflow WRITE 8\nCrash Address: 0x611000004088\nCrash State:\n  std::__1::vector<double, std::__1::allocator<double> >::__append\n  PCIDSK::ProjParmsFromText\n  PCIDSK::CPCIDSKVectorSegment::GetProjection\n  \nSanitizer: address (ASAN)\n\nRecommended Security Severity: High\n\nRegressed: https://oss-fuzz.com/revisions?job=libfuzzer_asan_gdal&range=201806180619:201806182015\nFixed: https://oss-fuzz.com/revisions?job=libfuzzer_asan_gdal&range=201806210630:201806220629\n\nReproducer Testcase: https://oss-fuzz.com/download?testcase_id=5198634147643392\n\nSee https://github.com/google/oss-fuzz/blob/master/docs/reproducing.md for more information.\n\nIf you suspect that the result above is incorrect, try re-doing that job on the test case report page."
            },
            {
                "projectName": "oss-fuzz",
                "localId": 8963,
                "sequenceNum": 5,
                "commenter": {
                    "userId": "382749006",
                    "displayName": "ClusterFuzz-External"
                },
                "timestamp": 1529651553,
                "content": "ClusterFuzz testcase 5198634147643392 is verified as fixed, so closing issue as verified.\n\nIf this is incorrect, please file a bug on https://github.com/google/oss-fuzz/issues/new",
                "amendments": [
                    {
                        "fieldName": "Status",
                        "newOrDeltaValue": "Verified",
                        "oldValue": "New"
                    },
                    {
                        "fieldName": "Labels",
                        "newOrDeltaValue": "ClusterFuzz-Verified"
                    }
                ]
            },
            {
                "projectName": "oss-fuzz",
                "localId": 8963,
                "sequenceNum": 6,
                "commenter": {
                    "userId": "4164592774",
                    "displayName": "sheriffbot@chromium.org"
                },
                "timestamp": 1532272921,
                "content": "This bug has been fixed for 30 days. It has been opened to the public.\n\n- Your friendly Sheriffbot",
                "amendments": [
                    {
                        "fieldName": "Labels",
                        "newOrDeltaValue": "-restrict-view-commit"
                    }
                ]
            },
            {
                "projectName": "oss-fuzz",
                "localId": 8963,
                "sequenceNum": 7,
                "commenter": {
                    "userId": "2965673606",
                    "displayName": "metzman@chromium.org"
                },
                "timestamp": 1551718590,
                "amendments": [
                    {
                        "fieldName": "Labels",
                        "newOrDeltaValue": "Security_Severity-High"
                    }
                ]
            }
        ]
    },
    "fix_commit": "c368014547ab9f9d7335254c02a87e08a156942a",
    "repo_addr": "https://github.com/OSGeo/gdal"
}